rootfs/etc/nginx/lua/balancer.lua

```diff
@@ -92,16 +122,26 @@ function _M.init_worker()
 end
 
 function _M.call()
-  ngx_balancer.set_more_tries(1)
+  local phase = ngx.get_phase()
+
+  if phase == "balancer" then
+    ngx_balancer.set_more_tries(1)
 
-  local host, port = balance()
+    local host, port = balance()
 
-  local ok
-  ok, err = ngx_balancer.set_current_peer(host, port)
-  if ok then
-    ngx.log(ngx.INFO, "current peer is set to " .. host .. ":" .. port)
+    local ok, err = ngx_balancer.set_current_peer(host, port)
+    if ok then
+      ngx.log(
+        ngx.INFO, 
+        "current peer is set to " .. host .. ":" .. port .. " using lb_alg " .. tostring(get_current_lb_alg())
+      )
+    else
+      ngx.log(ngx.ERR, "error while setting current upstream peer to: " .. tostring(err))
+    end
+  elseif phase == "log" then
```

Comment:
style: instead of using `if .. elseif` construct please use something like
```
if phase == "log" then
  after_balance()
  return
end

if phase ~= "balancer" then
  error('must be called in balancer or log, but was called in: ' .. phase)
  return
end

...
```